core/arch/arm/kernel/generic_boot.c

```diff
@@ -812,6 +812,10 @@ static void init_fdt(unsigned long phys_fdt)
 	dt_blob_addr = fdt;
 }
 
+static void carveout_memory_range(void)
```

Comment:
While renaming this function, maybe there is something to do here. Case no nsec memory info in DT (see `config_nsmem()`), this should discover nsec ddr if there is registered 'ddr_overall' memories (and carve them out).

This simplist seem to use the same sequence for `carveout_memory_range()` for cases `CFG_DT=y/n` and start by:
```
	if (!get_discovered_nsec_ddr(&dummy_start, &dummy_end))
 	 	discover_nsec_memory();
```

Reply:
Yes, there are indeed something some common process for nsec memory.
I will update a patch for this first, then we can discuss it in detail.

Reply:
By the way, I think the name of `init_fdt()` do not make sense. It only for DT feature, but it also process the memory. I think we should use a more common name here, such as "configure_cfg()" or some other name. So that we can merge the process of `carveout_memory_range()` into it.
What do you think? @etienne-lms

Reply:
`init_fdt()` is related to configuration specifically related to a non-secure DTB: op-tee reads some configs and writes some configs. May cover several domains (memory, tee api, ...). I would rather keep it with its scope.

Reply:
@etienne-lms +1 and btw `configure_console_from_dt(fdt)` should probably be moved from `init_primary_helper()` to `init_fdt()`.

Reply:
Ok, I understand.